src/api/rpc_api/author.rs

```diff
@@ -129,14 +153,272 @@ where
 	}
 }
 
-fn get_maybe_block_hash<Hash, BlockHash>(
-	transcation_status: TransactionStatus<Hash, BlockHash>,
-) -> Option<BlockHash> {
-	match transcation_status {
-		TransactionStatus::InBlock(block_hash) => Some(block_hash),
-		TransactionStatus::Retracted(block_hash) => Some(block_hash),
-		TransactionStatus::FinalityTimeout(block_hash) => Some(block_hash),
-		TransactionStatus::Finalized(block_hash) => Some(block_hash),
-		_ => None,
+impl<Signer, Client, Params, Runtime> SubmitAndWatchUntilSuccess<Client, Runtime::Hash>
+	for Api<Signer, Client, Params, Runtime>
+where
+	Client: Subscribe + Request,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+	Runtime::Hashing: HashTrait<Output = Runtime::Hash>,
+	Runtime: FrameSystemConfig + GetRuntimeBlockType,
+	Runtime::RuntimeBlock: BlockTrait + DeserializeOwned,
+	Runtime::Hashing: HashTrait<Output = Runtime::Hash>,
+	Runtime::Hash: FromHexString,
```

Comment:
I see us repeatedly having these excessive trait bounds. I would vote for adding them to the respective pallet's config, so we don't need to repeat them here every time.

Reply:
https://github.com/scs/substrate-api-client/issues/406